### DISCUSSION TITLE
How to properly pass register to children component

### DISCUSSION THREAD

**Discussion by STR4NG3R** [react-hook-form/react-hook-form#8938] [General] [answered]
I am currently working on react native personal project, and I am passing register as prop to children component, with some others props

![image](https://user-images.githubusercontent.com/13526621/187054988-0ca718aa-4991-450b-8182-ad42ad13cd23.png)

When i click the submit button the values are not changed, as below.
![image](https://user-images.githubusercontent.com/13526621/187055040-ea0d35dd-37b7-45a6-959b-7dded84d1a4e.png)

I don't know what i am doing wrong.

**Comment by Moshyfawn** [comment, marked as answer]
RN inputs have different props to the native web inputs the `register` method returns.

As a rule of thumb, it's advised to opt-in to use controlled components instead because of the nature of RN inputs _(controlled)_

**Reply by STR4NG3R** [reply]
I will let this code example:
![image](https://user-images.githubusercontent.com/13526621/187056169-3f429e4e-5b26-45cc-8cb3-02e00cef6628.png)

Thanks :)

**Reply by Moshyfawn** [reply]
Nice, thank you for sharing your example!
